### DISCUSSION TITLE
Could not authenticate: Device is not MikroTik

### DISCUSSION THREAD

**Discussion by MiguelPol** [eduardogsilva/routerfleet#47] [General]
Thank you very much for sharing your great work
I am using your tool, when I try to add several Mikrotik devices I get the following message - Could not authenticate: Device is not MikroTik
From putty I can access the mikrotik without any problems
I don't know what could be causing this message, since the SSH access is correct and the user has all the permissions. On other MikroTik it works without problems.
thank you

**Comment by eduardogsilva** [comment]
Sorry, I didn't noticed that the time passed so fast.

Can you please run a "/system resource print" and paste the output here?
When you try to authenticate, routerfleet will ssh to your system and run this command to check if you are configuring a mikrotik

Cheers

**Reply by MiguelPol** [reply]
Thanks for the reply


 /system resource print
             uptime: 10w3d7h9m38s
            version: 6.44.5 (long-term)
         build-time: Jul/04/2019 10:32:21
        free-memory: 1662.6MiB
       total-memory: 1952.0MiB
                cpu: tilegx
          cpu-count: 16
      cpu-frequency: 1200MHz
           cpu-load: 1%
     free-hdd-space: 79.9MiB
    total-hdd-space: 128.0MiB
  architecture-name: tile
         board-name: CCR1016-12S-1S+

**Reply by eduardogsilva** [reply]
I don't understand. It's missing a line: "platform: MikroTik"

Any idea of why?

**Reply by mjpol** [reply]
Good morning, I have been informed and some equipment was customized with the option of  Branding, the customization included the change of platform: Mikrotik to Prophase....

uptime: 10w3d23h29m39s
version: 6.44.5 (long-term)
build-time: Jul/04/2019 10:32:21
free-memory: 1657.7MiB
total-memory: 1952.0MiB
cpu: tilegx
cpu-count: 16
cpu-frequency: 1200MHz
cpu-load: 3%
free-hdd-space: 79.9MiB
total-hdd-space: 128.0MiB
architecture-name: tile
board-name: CCR1016-12S-1S+
**platform: Prophase**

**Reply by eduardogsilva** [reply]
Do you have any idea of what else is customized and what change on the OS? 
This is new for me, and I couldn't find any extra documentation about this.

**Reply by MiguelPol** [reply]
Hello, Mikrotik gives an option to customers to customize their equipment.

https://help.mikrotik.com/docs/spaces/ROS/pages/69664784/Branding

**Reply by eduardogsilva** [reply]
Well, I've found this https://help.mikrotik.com/docs/spaces/ROS/pages/69664784/Branding 
It doesn't looks like the branding could affect the backup process. 

I can fix this by removing "plataform: Mikrotik" from the verification and replacing by "plataform:" only.

The other option would be to create an extra dropdown option for "Mikrotik Branded" on the router configuration. I would need to adapt some parts of the code, but this would explicitly show that the router is different. This could help when someone needs support here.


Can you confirm if the plataform: Prophase was present and you removed on your first resource print?

Cheers

**Reply by MiguelPol** [reply]
I removed it myself, I assumed it was not such an important field, prophase is the name of our company

Either of the two options would work for me. I would really appreciate it if you could make that change because I see that most of the MikroTik that we have are modified with that parameter.

**Reply by eduardogsilva** [reply]
Great to hear that you are from Prophase. 
Can you share more details of the branding and customization permissions (not from Prophase)?

For what I read, I didn't see any customization option that could break backups. 
Is it limited to small details only, or does it have extra options beyond this link that allows you to change more deep settings like read/write permissions or other things like that?

(I will add an option to allow the branded routers, I'm just deciding the best way) 

Cheers

**Reply by MiguelPol** [reply]
We only have this customization in older routers and they are visually similar to what I have seen. It seems that it was some kind of test but in the end we did not see much use for it. Since it is in production, I do not want to generate a package again just to change that value to mikrotik

**Reply by eduardogsilva** [reply]
I've created an [issue](https://github.com/eduardogsilva/routerfleet/issues/52)
I will take a look on this soon.

Cheers!

**Reply by MiguelPol** [reply]
thank you so much

**Comment by eduardogsilva** [comment]
@MiguelPol if you upgrade to the latest version, you will be able to add Mikrotik Branded routers.

Let me know how it goes.
Cheers!!

**Reply by MiguelPol** [reply]
Hi, I have updated and it works correctly
Thank you very much
